provider/kubernetes_test.go

```diff
@@ -1524,11 +1525,283 @@ func TestServiceAnnotations(t *testing.T) {
 	}
 }
 
+func TestKubeAPIErrors(t *testing.T) {
+	ingresses := []*v1beta1.Ingress{{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "testing",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "foo",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/bar",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "service1",
+										ServicePort: intstr.FromInt(80),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}}
+
+	services := []*v1.Service{{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "service1",
+			UID:       "1",
+			Namespace: "testing",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "10.0.0.1",
+			Ports: []v1.ServicePort{
+				{
+					Port: 80,
+				},
+			},
+		},
+	}}
+
+	endpoints := []*v1.Endpoints{}
+	watchChan := make(chan interface{})
+	apiErr := errors.New("failed kube api call")
+
+	testCases := []struct {
+		desc            string
+		apiServiceErr   error
+		apiEndpointsErr error
+	}{
+		{
+			desc:          "failed service call",
+			apiServiceErr: apiErr,
+		},
+		{
+			desc:            "failed endpoints call",
+			apiEndpointsErr: apiErr,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			t.Parallel()
+
+			client := clientMock{
+				ingresses:         ingresses,
+				services:          services,
+				endpoints:         endpoints,
+				watchChan:         watchChan,
+				apiServiceError:   tc.apiServiceErr,
+				apiEndpointsError: tc.apiEndpointsErr,
+			}
+
+			provider := Kubernetes{}
+			if _, err := provider.loadIngresses(client); err != apiErr {
+				t.Errorf("Got error %v, wanted error %v", err, apiErr)
+			}
+		})
+	}
+}
+
+func TestMissingResources(t *testing.T) {
+	ingresses := []*v1beta1.Ingress{{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "testing",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "foo",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "service1",
+										ServicePort: intstr.FromInt(80),
+									},
+								},
+							},
+						},
+					},
+				},
+				{
+					Host: "bar",
```

Comment:
Suggestion: Could we change the host name to something like "host_with_missing_service" or place an according comment to indicate that the purpose of this Ingress resource test-wise is to not be found because of that reason?

It takes a bit of back of forth scrolling to understand the purpose of the individual objects this test is setting up. I think purposeful names / comments could alleviate that.